Cython/Compiler/ExprNodes.py

```diff
@@ -3557,9 +3563,19 @@ def calculate_result_code(self):
             else:
                 assert False, "unexpected base type in indexing: %s" % self.base.type
         elif self.base.type.is_cfunction:
+            template_indices = []
+            for param in self.type_indices:
+                if isinstance(param, PyrexTypes.CType):
```

Comment:
I think these should be wrapped in some kind of tree nodes, rather than sitting right next to nodes in the same mixed-types list.